core/libs/shared/src/core_ops.rs

```diff
@@ -95,7 +93,23 @@ where
             if self.link(&id)?.is_some() {
                 continue;
             }
+            user_visible_ids.push(id);
+        }
+        self.resolve_and_finalize_all(account, user_visible_ids.into_iter(), public_key_cache)
+    }
+
+    pub fn resolve_and_finalize_all<I, PublicKeyCache>(
```

Comment:
when resolving files for the purposes of reporting file names of calculated work, we don't want to skip e.g. deleted files